OpenRA.Game/Widgets/SpriteWidget.cs

```diff
@@ -50,13 +51,16 @@ public override void Draw()
 		{
 			var sprite = GetSprite();
 			var palette = GetPalette();
+			var scale = GetScale();
+
+			var size = new float2(sprite.Size.X * scale, sprite.Size.Y * scale);
 
 			if (sprite == null || palette == null)
 				return;
 
 			if (sprite != cachedSprite)
 			{
-				offset = 0.5f * (new float2(RenderBounds.Size) - sprite.Size);
+				offset = 0.5f * scale * (new float2(RenderBounds.Size) - sprite.Size);
```

Comment:
The offset depends on both scale and sprite, but is only updated when sprite changes.  The offset will be incorrect if the GetScale returns a different result.